src/server/client.rs

```diff
@@ -223,9 +223,56 @@ async fn handle_autoalloc_message(
         AutoAllocRequest::AddQueue(params) => create_queue(server_dir, state_ref, params).await,
         AutoAllocRequest::Events { descriptor } => get_event_log(state_ref, descriptor),
         AutoAllocRequest::Info { descriptor } => get_allocations(state_ref, descriptor),
+        AutoAllocRequest::RemoveQueue(id) => remove_queue(state_ref, id).await,
     }
 }
 
+async fn remove_queue(state_ref: &StateRef, id: DescriptorId) -> ToClientMessage {
+    let remove_alloc_fut = {
+        let mut server_state = state_ref.get_mut();
+        let descriptor_state = server_state
+            .get_autoalloc_state_mut()
+            .get_descriptor_mut(id);
+
+        match descriptor_state {
+            Some(state) => {
+                let handler = state.descriptor.handler();
+                let futures: Vec<_> = state
+                    .active_allocations()
+                    .map(move |alloc| {
+                        let future = handler.remove_allocation(alloc.id.clone());
+                        async move { (future.await, alloc.id.clone()) }
```

Comment:
You probably removed the `id` local variable here, so you were referencing the parameter instead of a local variable. But the variable is needed, otherwise the `async move` future will hold a reference to state, which is a problem.

It should look like this:
```rust
let future = handler.remove_allocation(alloc.id.clone());
let id = alloc.id.clone();
async move { (future.await, id) }
```

Reply:
Thank you,  I need to learn to see these